modules/imgcodecs/test/test_read_write.cpp

```diff
@@ -303,4 +303,97 @@ TEST(Imgcodecs_Image, write_umat)
     EXPECT_EQ(0, remove(dst_name.c_str()));
 }
 
+TEST(ImgCodecs_Imread, read_signature_fail)
+{
+    const string root = cvtest::TS::ptr()->get_data_path();
+    const string filename = root + "../highgui/pngsuite/xcrn0g04.png";
+
+    cv::Mat m;
+    ImreadParams params;
+    params.flags = IMREAD_COLOR;
+    ImreadError err = imread2(filename, m, params);
+    EXPECT_EQ(err, IMREAD_ERROR_UNRECOGNIZED_FORMAT);
```

Comment:
Signature for `_EQ` is `ASSERT_EQ(expected_reference, actual_result);`.
Correct order of parameters are required to emit valid error messages.

Reference: https://github.com/opencv/opencv/blob/4.0.0/modules/ts/include/opencv2/ts/ts_gtest.h#L8196-L8200

```
GTEST_API_ AssertionResult EqFailure(const char* expected_expression,
                                     const char* actual_expression,
                                     const std::string& expected_value,
                                     const std::string& actual_value,
                                     bool ignoring_case);
```